include/libtorrent/session_handle.hpp

```diff
@@ -1014,6 +1014,18 @@ namespace libtorrent {
 		port_mapping_t add_port_mapping(portmap_protocol t, int external_port, int local_port);
 		void delete_port_mapping(port_mapping_t handle);
 
+		// Instructs the session to reopen all listen and outgoing sockets.
+		//
+		// It's useful in the case your platform doesn't support the built in
+		// IP notifier mechanism, or if you have a better more reliable way to
+		// detect changes in the IP routing table.
+		//
+		// The ``map_ports`` indicates if the ports are mapped using natpmp
+		// and upnp. If mapping was already made, they are deleted and added
+		// again. This only works if natpmp and/or upnp are configured to be
+		// enable.
+		void reopen_network_sockets(bool map_ports = true);
```

Comment:
instead of taking a bool here, I think you should take flags. Specifically a "type-safe" flag. The benefits are extensibility with preserved ABI, easier to read call sites and stronger type safety

Reply:
ok, will do that